api/envoy/service/discovery/v2/csds.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+package envoy.service.discovery.v2;
+
+option java_package = "io.envoyproxy.envoy.service.discovery.v2";
+option java_outer_classname = "CsdsProto";
+option java_multiple_files = true;
+option java_generic_services = true;
+
+import "envoy/admin/v2alpha/config_dump.proto";
+import "envoy/api/v2/core/base.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+// CSDS is Client Status Discovery Service. It can be used to get the status of
+// xDS-compliant client from the management server's point of view. It can also
+// be used as an interface to get the current state of the client.
+service ClientStatusService {
+  rpc StreamClientStatus(stream ClientStatusRequest) returns (stream ClientStatusResponse) {
```

Comment:
Why is this not DiscoveryRequest/Response? I though this was a first-class xDS protocol.

Reply:
As discussed offline, the client status request can include a list of clients. Would like to keep the current definition.